hazelcast/src/main/java/com/hazelcast/map/impl/operation/BasePutOperation.java

```diff
@@ -50,36 +47,15 @@ public BasePutOperation() {
     @Override
     public void afterRun() {
         mapServiceContext.interceptAfterPut(name, dataValue);
-        Object value = isPostProcessing(recordStore) ? recordStore.getRecord(dataKey).getValue() : dataValue;
+        Record record = recordStore.getRecord(dataKey);
```

Comment:
Now we unconditionally get the record. Could this affect performance if the record store is not post-processing?

Reply:
seems negligible to me.